eZ/Publish/Core/FieldType/TextLine/Type.php

```diff
@@ -64,9 +64,9 @@ public function validateValidatorConfiguration($validatorConfiguration)
                 switch ($name) {
                     case 'minStringLength':
                     case 'maxStringLength':
-                        if ($value !== null && !is_integer($value)) {
+                        if ($value !== null && (!is_integer($value) || $value < 0)) {
```

Comment:
two tests? i don't understand you hear. 
and btw, if we plan to add more improvements in this issue as you just commented, maybe we should change everything a bit?
we need to validate three things
 * min and maxLenghts are positives
 * max are bigger or equal than min
 * default value lenght is greater or equal than min and not greater than max